File: android/src/main/kotlin/com/eyu/eyu_open_sdk_flutter/AdMessageCodec.java

package com.eyu.eyu_open_sdk_flutter;


import android.content.Context;

import androidx.annotation.NonNull;

import com.eyu.opensdk.ad.EyuAd;
import com.eyu.opensdk.ad.base.model.LoadAdError;

import java.io.ByteArrayOutputStream;

import io.flutter.plugin.common.StandardMessageCodec;

class AdMessageCodec extends StandardMessageCodec {
  private static final byte VALUE_AD = (byte) 128;
  private static final byte VALUE_AD_ERROR = (byte) 129;

  @NonNull
  final Context context;

  AdMessageCodec(@NonNull Context context) {
    this.context = context;
  }

  @Override
  protected void writeValue(ByteArrayOutputStream stream, Object value) {
    if (value instanceof EyuAd) {
      stream.write(VALUE_AD);
      final EyuAd request = (EyuAd) value;
      writeValue(stream, request.getUnitId());
      writeValue(stream, request.getPlaceId());
      writeValue(stream, request.getAdFormat().getLabel());
      writeValue(stream, request.getAdRevenue());
      writeValue(stream, request.getMediator());
      writeValue(stream, request.getNetworkName());
    }else if(value instanceof LoadAdError){
      stream.write(VALUE_AD_ERROR);
      final LoadAdError error = (LoadAdError) value;
      writeValue(stream, error.getCode());
      writeValue(stream, error.getDisplayMessage());
    }else {
      super.writeValue(stream, value);
    }
  }

}

File: android/src/main/kotlin/com/eyu/eyu_open_sdk_flutter/AdMessageCodec.kt
package com.eyu.eyu_open_sdk_flutter

import android.content.Context
import com.eyu.opensdk.ad.EyuAd
import com.eyu.opensdk.ad.base.model.LoadAdError
import io.flutter.plugin.common.StandardMessageCodec
import java.io.ByteArrayOutputStream

internal class AdMessageCodec(val context: Context) : StandardMessageCodec() {
    companion object {
        private const val VALUE_AD = 128.toByte()
        private const val VALUE_AD_ERROR = 129.toByte()
    }

    override fun writeValue(stream: ByteArrayOutputStream, value: Any) {
        if (value is EyuAd) {
            stream.write(VALUE_AD.toInt())
            val request = value
            writeValue(stream, request.unitId)
            writeValue(stream, request.placeId)
            writeValue(stream, request.adFormat.label)
            writeValue(stream, request.adRevenue)
            writeValue(stream, request.mediator)
            writeValue(stream, request.networkName)
        } else if (value is LoadAdError) {
            stream.write(VALUE_AD_ERROR.toInt())
            val error = value
            writeValue(stream, error.code)
            writeValue(stream, error.displayMessage)
        } else {
            super.writeValue(stream, value)
        }
    }


}